test/parallel/test-dotenv-node-options.js

```diff
@@ -0,0 +1,98 @@
+    assert.strictEqual(child.code, 0);
+  });
+
+  it('validate only read permissions are enabled', async () => {
+    const code = `
+      "require('fs').writeFileSync(require('path').join(__dirname, 'should-not-write.txt'), 'hello', 'utf-8')"
+    `.trim();
+    const child = await spawnPromisified(process.execPath,
+                                         [ `--env-file=${relativePath}`, '--eval', code ], {
+                                           cwd: __dirname,
+                                         });
+    assert.strictEqual(child.code, 0);
+  });
+
+  it('TZ environment variable', async () => {
+    const code = `
+      "require('assert').assert(new Date().toString().includes('Hawaii'))"
+    `.trim();
+    const child = await spawnPromisified(process.execPath,
+                                         [ `--env-file=${relativePath}`, '--eval', code ], {
+                                           cwd: __dirname,
+                                         });
+    assert.strictEqual(child.stderr, '');
+    assert.strictEqual(child.code, 0);
+  });
+
+  it('should update UV_THREADPOOL_SIZE', async () => {
+    const code = `
+      "require('assert').assert.strictEqual(process.env.UV_THREADPOOL_SIZE, 5)"
+    `.trim();
+    const child = await spawnPromisified(process.execPath,
+                                         [ `--env-file=${relativePath}`, '--eval', code ], {
+                                           cwd: __dirname,
+                                         });
+    assert.strictEqual(child.stderr, '');
+    assert.strictEqual(child.code, 0);
+  });
+
+  it('should support absolute paths', async () => {
+    const code = `
+      "require('assert').assert.strictEqual(process.env.CUSTOM_VARIABLE, 'hello-world')"
+    `.trim();
+    const child = await spawnPromisified(process.execPath,
+                                         [ `--env-file=${absolutePath}`, '--eval', code ], {
+                                           cwd: __dirname,
+                                         });
+    assert.strictEqual(child.stderr, '');
+    assert.strictEqual(child.code, 0);
+  });
+
+  it('should use the last --env-file declaration', async () => {
+    const code = `
+      "require('assert').strictEqual(process.env.BASIC, 'basic')"
+    `.trim();
+    const child = await spawnPromisified(process.execPath,
+                                         [ `--env-file=${relativePath}`, `--env-file=${validEnvFilePath}`, '--eval', code ], {
+                                           cwd: __dirname,
+                                         });
+    assert.strictEqual(child.stderr, '');
+    assert.strictEqual(child.code, 0);
+  });
+
+  it('should handle unexisting .env file', async () => {
+    const code = `
+      "require('assert').strictEqual(1, 1)"
+    `.trim();
+    const child = await spawnPromisified(process.execPath,
+                                         [ '--env-file=.env', '--eval', code ], {
```

Comment:
I would expect this to throw

Reply:
I don't have any strong opinion on this, but: this is the default behavior in dotenv.

Reply:
Resolving this in favor the issue I've created.